spring-integration-core/src/test/java/org/springframework/integration/support/leader/LockRegistryLeaderInitiatorTests.java

```diff
@@ -76,17 +78,40 @@ public void init() {
 	}
 
 	@Test
-	public void startAndStop() throws Exception {
-		assertThat(this.initiator.getContext().isLeader(), is(false));
-		this.initiator.start();
-		assertThat(this.initiator.isRunning(), is(true));
+	public void startAndStopWithRegistryAndCandidate() throws Exception {
+		startStopAndVerifyInitiatorLeadership(this.initiator);
+	}
+
+	@Test
+	public void startAndStopWithRegistryCandidateAndExecutorService() throws Exception {
+		startStopAndVerifyInitiatorLeadership(
+				new LockRegistryLeaderInitiator(this.registry,
+						new DefaultCandidate(),
+						Executors.newSingleThreadExecutor(new CustomizableThreadFactory("lock-leadership-")))
+		);
+	}
+
+	@Test
+	public void startAndStopWithRegistryAndExecutorService() throws Exception {
```

Comment:
I think your new tests don't bring any value.

Would be better to add setter, set an external executor, call `destroy()` on the `LockRegistryLeaderInitiator` and in the end ensure that exactly this external executor is used. Plus its `shutdown()` hasn't been called.

TO repeat a test-case with the same logic doesn't expose any vulnerability.

Reply:
Fair points, will make the changes and was just trying to cover my changes.  Certainly the destroy() and shutdown() verifies make sense.  will make changes and update PR

Reply:
Good. Thank you very much for the update! I will come to you for review tomorrow. 

For this particular case to check an internal executor you can consider to use `TestUtils.getPropertyValue()` where we use reflection on the matter

Reply:
Ok awesome. I’ll add the test case and then push, most likely tomorrow